### PR TITLE
Create an easy to install distribution for raiden #179

### PR DESCRIPTION
**Pull request by konradkonrad**
Goals:
- [x] GNU/Linux distributable
- [x] ~Mac OS/X distributable~ out of scope for now
- [x] ~Windows executable~ out of scope for now
- [x] add proper versioning (use `ARG version` in `build.Dockerfile` to control git checkout, paste version into `.desktop` file)
- [x] survey what Glib minimum version we want to support (currently `2.14`)  

For review here a link to an AppImage from git commit 9b26fe486a0edf9cff93d73d0d5a2f369743ea39 created via `make bundle`: https://www.sendspace.com/file/xv33fd

#### What's in the PR?

The PR adds a couple of `make` targets:
- `make bundle` -- create a linux AppImage and store it in `dist/raiden--x86_64.AppImage`
- `make test-bundle` -- test that the AppImage launches on several linux distributions
- `make test-bundle-unsupported` -- test distributions where launching the AppImage fails (due to `Glib < 2.14`)

The `make bundle` target creates an [AppImage](https://github.com/probonopd/AppImageKit) with bundled python 2.7.12. AppImages are self-contained disk images with a custom launcher that adjusts the environment for the bundled application+libraries and launches the bundled application. Executing the AppImage will directly start raiden, no further installation required.

The only requirements (as far as I know) are:
- `fuse` libraries installed & fuse permissions for the user (needed for the launcher to mount the image)
- `glib` with a version `>=2.14`
- for running `raiden` a user will still need to have access to an Ethereum jsonrpc endpoint
- `solc` is NOT required, thanks to #428 the bundle contains precompiled contract ABIs

The image creation is scripted in a newly created Dockerfile, `docker/build.Dockerfile`. This ensures that
- we should be able to reproduce the builds independent of the build host
- no build host context spills into the bundled AppImage

Currently supported distributions (from their docker image):
- ubuntu:17.04
- ubuntu:16.10
- ubuntu:16.04
- ubuntu:14.04
- debian:8
- debian:9
- base/archlinux
- centos:7
- fedora:20
- fedora:21
- fedora:22
- fedora:23
- fedora:24
- fedora:25
- fedora:rawhide

not working on these:
- centos:5
- centos:6
- debian:7

### CODE REVIEW

**Comment by konradkonrad** [comment]
I'll put this on hold until #430 is resolved.

File: docker/build.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# backward compatibility:
+FROM ubuntu:12.04.5
```
Comment (ulope) [inline docker/build.Dockerfile]: Does it make sense to still use 12.04 here? It will be EOL in April: https://wiki.ubuntu.com/Releases

Comment (probonopd) [inline docker/build.Dockerfile]: The older the better, this way it also runs on older versions of e.g., debian and CentOS and RHEL and whatnot. But it is up to you to decide which target systems you still want to support. Some application projects build on CentOS 6.x for this reason.

File: docker/raiden.desktop [hunk 2]
```diff
@@ -0,0 +1,7 @@
+[Desktop Entry]
+Name=Raiden
+Version=XXVERSIONXX
+Exec=raiden
+Icon=raiden
```
Comment (konradkonrad) [inline docker/raiden.desktop]: As far as I understand AppImage specification wants an icon. The file is just an svg (so no blob). So I kindly ask you to disregard that comment :P

Comment (LefterisJP) [inline docker/raiden.desktop]: yeah forgot svg is in xml, so it's fine

File: requirements.txt [hunk 3]
```diff
@@ -2,8 +2,8 @@ pysha3
 # temporary until new version of pyethereum is released, that supports solc >= v0.4.9
 -e git+https://github.com/LefterisJP/pyethapp@use_new_solc_combinedjson_key#egg=pyethapp
 ipython<5.0.0
-rlp>=0.4.3,<=0.4.6
```
Comment (konradkonrad) [inline requirements.txt]: This is also in #443 -- I'll wait for that and take it out of my PR.

Comment (LefterisJP) [inline requirements.txt]: cool, I see.

File: docker/build.Dockerfile (hunk 1 above)
Comment (ulope) [inline docker/build.Dockerfile]: > The older the better

In principle yes but 12.04 will be EOL next month which means that any new vulnerabilities which might crop up in the future won't be patched anymore. Thereby we could end up shipping possibly exploitable code inside the AppImage.

**Comment by konradkonrad** [comment]
this closed #179

**Pull request merged**